decorator.rb

```diff
@@ -0,0 +1,12 @@
+require_relative './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable.correct_name
+  end
+end
```

Comment:
- [ ] The nameable property needs to be readable and writable. We, therefore, need to add: `attr_accessor :nameable` before the initialize block. Please do this to ensure the class can usable.